### PR TITLE
9635 - Add description fields to most important editor components

### PR DESCRIPTION
**Pull request by Cattlesquat**
#9635

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/build/Widget.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
  */
 package VASSAL.build;
 
+import VASSAL.configure.ComponentDescription;
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/Widget.java:20]: Group with the other `VASSAL` imports.

File: vassal-app/src/main/java/VASSAL/build/module/turn/TurnComponent.java [hunk 2]
```diff
@@ -26,13 +26,22 @@
...
-public class TurnComponent extends AbstractConfigurable {
+public class TurnComponent extends AbstractConfigurable implements ComponentDescription {
+  public static final String DESCRIPTION = "description"; //NON-NLS
+
+  protected String description;
+
+  @Override
+  public String getDescription() {
+    return description;
+  }
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/turn/TurnComponent.java:44]: What sets it?

**Review by uckelman — CHANGES_REQUESTED** [review]

File: vassal-app/src/main/java/VASSAL/build/module/turn/TurnComponent.java (hunk 2 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/turn/TurnComponent.java:44]: It gets set by the setAttribute()

File: vassal-app/src/main/java/VASSAL/build/Widget.java (hunk 1 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/Widget.java]: fixed

File: vassal-app/src/main/java/VASSAL/build/module/turn/TurnComponent.java (hunk 2 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/turn/TurnComponent.java:44]: (and I should explain that TurnComponent, even though it isn't declared abstract, is ONLY overridden by the various fleshed-out kinds of turn component)

Comment (uckelman) [reply vassal-app/src/main/java/VASSAL/build/module/turn/TurnComponent.java:44]: The horror, the horror. Ok, np.

**Review by uckelman — APPROVED** [review]

**Pull request merged**